### ISSUE TITLE
Update SUBQL Loans Info to reflect loan pallet updates

### ISSUE THREAD

**Issue by filo87** [embrio-tech/centrifuge-subql#150]
### User story

> **As** (who?) a data consumer
**I want** (what?) loan information to be indexed correctly from the new loan pallet implementation
**So that** (why?) i can leverage and show the data

### Types and Methods to Verify
in cfg-development/centrifuge-subql/src/mappings/services/poolService.ts
- [x] [api.query.loans.portfolioValuation](https://github.com/embrio-tech/centrifuge-subql/blob/9bab7d47b8e4d80706871a41f5e7c40160824003/src/mappings/services/poolService.ts#L93)
- [x] [api.query.loans.activeLoans](https://github.com/embrio-tech/centrifuge-subql/blob/9bab7d47b8e4d80706871a41f5e7c40160824003/src/mappings/services/poolService.ts#L166)

in: https://github.com/embrio-tech/centrifuge-subql/blob/150-Update-SUBQL-Loans-Info-to-reflect-loan-pallet-updates/src/mappings/handlers/loansHandlers.ts
- [x] LoanBorrowedRepaidEvent
- [x] LoanClosedEvent
- [x] LoanCreatedEvent
- [x] LoanWrittenOffEvent 

### Acceptance criteria

- [x] All Loan types match the current pallet implementation
- [x] Loans are indexed correctly and snapshot at regular intervals
- [x] Portfolio valuation is indexed correctly and snapshot for each pool
- [x] Tests updated and passing
- [x] SubQL entities for Loan and snapshots are updated accordingly

**Comment by filo87** [comment]
Meeting on monday 1600 to walkthrough of code

**Comment by filo87** [comment]
```bash
cfg-development-subql-node-1  | 2023-07-17T13:45:07.008Z <sandbox> INFO Querying active loan data for pool: 442507375 
cfg-development-subql-node-1  | 2023-07-17T13:45:07.029Z <sandbox> INFO loanDetails: [[5,{"schedule":{"maturity":{"fixed":1701302400},"interestPayments":"None","payDownSchedule":"None"},"collateral":[2727401731,"0x000000000000000044a06836f531314b"],"restrictions":{"borrows":"NotWrittenOff","repayments":"None"},"borrower":"kALowUVzu4obELgpc4Hrgb3E2LFbzmwiuscQwbbkAKgehNVTT","writeOffPercentage":0,"originationDate":1688639502,"pricing":{"external":{"info":{"priceId":{"isin":"0x55533931323739365a443432"},"maxBorrowAmount":{"noLimit":null}},"outstandingQuantity":1967000000}},"totalBorrowed":194369940000,"totalRepaid":1962000000,"totalRepaidUnchecked":0}], [12,{"schedule":{"maturity":{"fixed":1703116800},"interestPayments":"None","payDownSchedule":"None"},"collateral":[2727401731,"0x00000000000000003ff3abe826e1e67d"],"restrictions":{"borrows":"NotWrittenOff","repayments":"None"},"borrower":"kALowUVzu4obELgpc4Hrgb3E2LFbzmwiuscQwbbkAKgehNVTT","writeOffPercentage":0,"originationDate":1689006114,"pricing":{"external":{"info":{"priceId":{"isin":"0x555339313237393746563433"},"maxBorrowAmount":{"noLimit":null}},"outstandingQuantity":10000000}},"totalBorrowed":985000000,"totalRepaid":0,"totalRepaidUnchecked":0}], [3,{"schedule":{"maturity":{"fixed":1694649600},"interestPayments":"None","payDownSchedule":"None"},"collateral":[2727401731,"0x000000000000000035cefd90d53020b4"],"restrictions":{"borrows":"NotWrittenOff","repayments":"None"},"borrower":"kALowUVzu4obELgpc4Hrgb3E2LFbzmwiuscQwbbkAKgehNVTT","writeOffPercentage":0,"originationDate":1688633238,"pricing":{"external":{"info":{"priceId":{"isin":"0x555339313237393643513032"},"maxBorrowAmount":{"noLimit":null}},"outstandingQuantity":1980000000}},"totalBorrowed":195930900000,"totalRepaid":0,"totalRepaidUnchecked":0}], [4,{"schedule":{"maturity":{"fixed":1697500800},"interestPayments":"None","payDownSchedule":"None"},"collateral":[2727401731,"0x00000000000000000d1fcd816a49af38"],"restrictions":{"borrows":"NotWrittenOff","repayments":"None"},"borrower":"kALowUVzu4obELgpc4Hrgb3E2LFbzmwiuscQwbbkAKgehNVTT","writeOffPercentage":0,"originationDate":1688639082,"pricing":{"external":{"info":{"priceId":{"isin":"0x555339313237393748423630"},"maxBorrowAmount":{"noLimit":null}},"outstandingQuantity":1870000000}},"totalBorrowed":184120200000,"totalRepaid":0,"totalRepaidUnchecked":0}], [6,{"schedule":{"maturity":{"fixed":1690848000},"interestPayments":"None","payDownSchedule":"None"},"collateral":[2727401731,"0x000000000000000065c0f877b9a7f25d"],"restrictions":{"borrows":"NotWrittenOff","repayments":"None"},"borrower":"kALowUVzu4obELgpc4Hrgb3E2LFbzmwiuscQwbbkAKgehNVTT","writeOffPercentage":0,"originationDate":1688642934,"pricing":{"external":{"info":{"priceId":{"isin":"0x555339313237393747413936"},"maxBorrowAmount":{"noLimit":null}},"outstandingQuantity":1750000000}},"totalBorrowed":174282500000,"totalRepaid":0,"totalRepaidUnchecked":0}], [9,{"schedule":{"maturity":{"fixed":1703116800},"interestPayments":"None","payDownSchedule":"None"},"collateral":[2727401731,"0x00000000000000000ae1a7fcc03b0302"],"restrictions":{"borrows":"NotWrittenOff","repayments":"None"},"borrower":"kALowUVzu4obELgpc4Hrgb3E2LFbzmwiuscQwbbkAKgehNVTT","writeOffPercentage":0,"originationDate":1688992698,"pricing":{"external":{"info":{"priceId":{"isin":"0x555339313237393746563433"},"maxBorrowAmount":{"noLimit":null}},"outstandingQuantity":0}},"totalBorrowed":990000000,"totalRepaid":980000000,"totalRepaidUnchecked":0}], [13,{"schedule":{"maturity":{"fixed":1703116800},"interestPayments":"None","payDownSchedule":"None"},"collateral":[2727401731,"0x00000000000000003a4aab6435667e4c"],"restrictions":{"borrows":"NotWrittenOff","repayments":"None"},"borrower":"kALowUVzu4obELgpc4Hrgb3E2LFbzmwiuscQwbbkAKgehNVTT","writeOffPercentage":0,"originationDate":1688998242,"pricing":{"external":{"info":{"priceId":{"isin":"0x555339313237393746563433"},"maxBorrowAmount":{"noLimit":null}},"outstandingQuantity":5000000}},"totalBorrowed":1460000000,"totalRepaid":990000000,"totalRepaidUnchecked":0}], [14,{"schedule":{"maturity":{"fixed":1692316800},"interestPayments":"None","payDownSchedule":"None"},"collateral":[2727401731,"0x00000000000000002684264a04096df8"],"restrictions":{"borrows":"NotWrittenOff","repayments":"None"},"borrower":"kALowUVzu4obELgpc4Hrgb3E2LFbzmwiuscQwbbkAKgehNVTT","writeOffPercentage":0,"originationDate":1689255126,"pricing":{"external":{"info":{"priceId":{"isin":"0x55533931323739365a443432"},"maxBorrowAmount":{"noLimit":null}},"outstandingQuantity":95000000}},"totalBorrowed":9850000000,"totalRepaid":495500000,"totalRepaidUnchecked":0}]] 
cfg-development-subql-node-1  | 2023-07-17T13:45:07.030Z <sandbox> ERROR undefined TypeError: Cannot read properties of undefined (reading 'toString')
```
